kaldb/src/main/java/com/slack/kaldb/metadata/cache/CacheSlotMetadata.java

```diff
@@ -22,8 +23,10 @@ public CacheSlotMetadata(
       Metadata.CacheSlotMetadata.CacheSlotState cacheSlotState,
       String replicaId,
       long updatedTimeEpochMs,
-      List<Metadata.IndexType> supportedIndexTypes) {
+      List<Metadata.IndexType> supportedIndexTypes,
+      String hostname) {
     super(name);
+    checkArgument(hostname != null && !hostname.isEmpty(), "Hostname cannot be null or empty");
```

Comment:
As cache slots are ephemeral, this should be safe to enforce in this PR as the code deploying will re-create the slots with the newly required hostname.